### PR TITLE
os/common,arch/stm32l4:Fix Build breaks on stm board

### PR DESCRIPTION
**Pull request by yaswanth278**
change variable name to _sint_heap_start to fix issue Fix mtd_dev_s has no member named 'mtd' by changing mtd.name to name as name is a priv member

### CODE REVIEW

File: os/arch/arm/src/common/up_internal.h [hunk 1]
```diff
@@ -278,7 +278,7 @@ EXTERN uint32_t _edata;			/* End+1 of .data */
 EXTERN uint32_t _sbss;			/* Start of .bss */
 EXTERN uint32_t _ebss;			/* End+1 of .bss */
 EXTERN uint32_t _sidle_stack;		/* Start of idle stack */
-EXTERN uint32_t __sint_heap_start;	/* Start of heap in internal RAM region */
+EXTERN uint32_t _sint_heap_start;	/* Start of heap in internal RAM region */
 EXTERN uint32_t __sext_heap_start;	/* Start of heap in external RAM region */
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_internal.h:282]: Need to check this as well.

File: os/arch/arm/src/stm32l4/stm32l4_ospi.c [hunk 2]
```diff
@@ -1921,7 +1921,7 @@ FAR struct mtd_dev_s *up_flashinitialize(void)
     priv->read = this_read;
     priv->ioctl = this_ioctl;
 #ifdef CONFIG_MTD_REGISTRATION
-    priv->mtd.name = "stm32l4_flash";
+    priv->name = "stm32l4_flash";
```
Comment (sunghan-chang) [inline os/arch/arm/src/stm32l4/stm32l4_ospi.c:1924]: This is not `Fix __sint_heap_start undeclared issue`. If this is necessary to success a build, let's title like `.....: fix build breaks on stm board`.

File: build/configs/stm32l4r9ai-disco/pm/defconfig
Comment (sunghan-chang) [inline build/configs/stm32l4r9ai-disco/pm/defconfig:1]: Is this necessary to make a build?

**Comment by amogh-samsung** [comment]
this is a comment

File: os/arch/arm/src/common/up_internal.h (hunk 1 above)
Comment (amogh-samsung) [inline os/arch/arm/src/common/up_internal.h:282]: _s as a prefix to any variable indicates that the variables denotes the starting address.
For ex: - _sidle_stack denotes the start of the idle stack.
An extra "_start" at the end is unnecessary, i.e. _sidle_stack_start is not needed.

Similarly, _sint_heap is enough to denote the start of the internal heap
The extra "_start" at the end is unnecessary.

**Comment by sunghan-chang** [comment]
I can't find any difference from two commit titles.
1. Build board by changing variable to _sint_heap
2. Build board by changing variable to _sint_heap_start

Let's make three commits
1. rename _sint_heap_start to _sint_heap, _sext_heap_start to _sext_heap
2. mtd related change
3. defconfig update

**Comment by sunghan-chang** [comment]
And you should follow TizenRT commit rules. It's the commit rules, not PR rules.

**Comment by amogh-samsung** [comment]
1. Please make a set of changes to fix the build issues on the STM board and push them with a commit name like :-
"<HCF of path>: Fix build breaks in stm board"
Please take a not of all comments on the PR while making these changes. Please analyse what changes are necessary to fix the build issue, and what changes aren't.
(Then, after applying this commit to your repository, make the following changes)
2. Please change references to _sint_heap_start -> _sint_heap, _eint_heap_end -> _eint_heap
Add these changes to a second commit and push as "Remove redundancy in heap start/end address variable names"

**Comment by amogh-samsung** [comment]
Your commits should look like this :- 

"<HCF of path of files changed>: Commit title
- <Expanded description of the title and the changes proposed in this commit in 1-2 sentences>"

**Comment by yaswanth278** [comment]
@sunghan-chang Actually iam making two commits
1.defconfig,MTD variable, changing variable to _sint_heap_start these three together are required to build the board..This is first commit
2.As _sint_heap_start is redundant which iam using to build commit..I changed all files in repo from _sint_heap_start to _sint_heap and these variable changes are added in second commit

And now I will add proper commit titles as per amoghs review..

File: build/configs/stm32l4r9ai-disco/pm/defconfig [hunk 3]
```diff
@@ -538,8 +538,9 @@ CONFIG_ARCH_BOARD_HAVE_FLASH=y
 #
 CONFIG_FLASH_PARTITION=y
```
Comment (Tirtha-Samsung) [inline build/configs/stm32l4r9ai-disco/pm/defconfig:539]: Change the title to os/arch/arm/src/common : Change the var name from this to that ti fix the build breaks or something

File: docs/HowToAddnewBoard.md [hunk 4]
```diff
@@ -220,8 +220,8 @@ In addition to the SECTION information, add the following variables as well to m
 1. `_sbss`: Start of .bss section
 2. `_ebss`: End+1 of .bss section
 3. `_sidle_stack`: Start of idle stack
-4. `__sint_heap_start`: Start of heap in internal RAM region
-5. `__sext_heap_start`: Start of heap in external RAM region
+4. `__sint_heap`: Start of heap in internal RAM region
+5. `__sext_heap`: Start of heap in external RAM region
```
Comment (amogh-samsung) [inline docs/HowToAddnewBoard.md:224]: Please use single underscore '_' in all places.

Comment (amogh-samsung) [reply docs/HowToAddnewBoard.md:224]: for `sint_heap` and `sext_heap`

**Comment by sunghan-chang** [comment]
@yaswanth278 @amogh-samsung With this PR, there is no more build break on stm? Could you confirm ?

File: os/arch/arm/src/common/up_internal.h [hunk 5]
```diff
@@ -278,7 +278,7 @@ EXTERN uint32_t _edata;			/* End+1 of .data */
 EXTERN uint32_t _sbss;			/* Start of .bss */
 EXTERN uint32_t _ebss;			/* End+1 of .bss */
```
Comment (amogh-samsung) [inline os/arch/arm/src/common/up_internal.h:279]: Please correct the path for the changes described in the commit title
"os/arch/arm/up_internal.h" -> "os/arch/arm/src/common"

**Comment by amogh-samsung** [comment]
> @yaswanth278 @amogh-samsung With this PR, there is no more build break on stm? Could you confirm ?

++ @Tirtha-Samsung

**Comment by sunghan-chang** [comment]
@yaswanth278 @amogh-samsung It takes tooo much time. Let's finish it ASAP.

**Comment by Tirtha-Samsung** [comment]
> @Tirtha-Samsung @amogh-samsung @yaswanth278 Please confirm there is no more build break with this PR on STM build.

There is no build break

**Comment by yaswanth278** [comment]
> @Tirtha-Samsung @amogh-samsung @yaswanth278 Please confirm there is no more build break with this PR on STM build.

It is working fine

**Comment by yaswanth278** [comment]
> @Tirtha-Samsung @amogh-samsung @yaswanth278 Please confirm there is no more build break with this PR on STM build.

@sunghan-chang There is no build break and made changes as per PR comments.can you please review this.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**